### DISCUSSION TITLE
Descriptor neuron vs fitting net neuron

### DISCUSSION THREAD

**Discussion by krishnapitike** [deepmodeling/deepmd-kit#1626] [Q&A]
Hello, could you please explain the difference between "neuron" fields in Descriptor and fitting net sections. Specifically:

1) The neuron field in fitting net section is set to equal values from inner to outer layer (for example [240, 240, 240]), where as in the case of descriptor neuron, the number of nodes increases from inner to outer layer (for example [10, 20, 40]). Could you please explain the reason?

2) I am fitting a model for a BCC metallic system with vacancies and a cutoff radius= 6.5 Angstroms. Is there a recommended number of layers for neurons of fitting net and descriptor? My vacancy formation energy is over-predicted by 200 meV with descriptor neuron=[25,50,100,200] and fitting net neuron=[240,240,240,240]. Would you recommend neuron architecture with greater number of nodes and layers?

3) The fitting step time is doubling if I double the number of nodes in descriptor ([25,50,100,200]->[50,100,200,400]) but not increasing if I double the number of nodes in descriptor neuron ([240,240,240,240]->[480,480,480,480]). Could you please help me understand this behavior?

**Comment by njzjz** [comment]
You are free to tune parameters by yourself. See also https://docs.deepmodeling.org/projects/deepmd/en/v2.1.0/troubleshooting/howtoset_netsize.html.

The timing of neural networks is also affected by the shape of inputs.

**Reply by krishnapitike** [reply]
Specifically, I want to understand the difference between Descriptor neuron vs fitting net neuron. There seems to be no explanation about this anywhere. Thanks in advance for your help.

**Reply by njzjz** [reply]
If you are talking about the mathematical expression, there is no difference between them.

**Comment by likefallwind** [comment]
If you understand Chinese, you can look this: https://www.bilibili.com/video/BV19a41187MD?spm_id_from=333.337.search-card.all.click    to understand this 2 networks.

If you don't know Chinese, I recommend you read the paper to have a better understanding.
